taiidani/terraform-provider-jenkins
# Add `display_name` support to `jenkins_folder` resource

# Dependencies

N/A

# What Is Changing

An optional `display_name` argument can now be set so that you can specify a custom name to display in the Jenkins UI for the folder.

# Test Steps

- [X] Have you updated the `docs/` folder to match your change?
- [X] Have you exercised your change using the `examples/` docker compose setup?

```sh
make test
go test -cover ./...
?       github.com/taiidani/terraform-provider-jenkins  [no test files]
ok      github.com/taiidani/terraform-provider-jenkins/jenkins  0.332s  coverage: 18.6% of statements
```

I tested the provider in my local Jenkins setup.
```
resource "jenkins_folder" "project" {
  name = "my-super-duper-project"
  display_name = "My Super Duper Project"
}
```

After applying, the change was present in the UI :)

<img width="457" alt="Screenshot 2021-07-10 at 17 49 53" src="https://user-images.githubusercontent.com/310030/125170508-3f235400-e1a7-11eb-830c-6fc1baf3f48b.png">


And when viewing the folder config page, I can see both name & display name values matching what I set in my Terraform code earlier.
<img width="557" alt="Screenshot 2021-07-10 at 17 50 12" src="https://user-images.githubusercontent.com/310030/125170516-49dde900-e1a7-11eb-9279-dcd379511aa2.png">
